lib/OpenQA/WebAPI/Controller/Admin/Needle.pm

```diff
@@ -44,6 +44,50 @@ sub _translate_cond($) {
     die "Unknown '$cond'";
 }
 
+sub _prepare_data_table {
+    my ($self, $n, $paths, $dir_rs, $needles_rs) = @_;
+    my $filename = $n->filename;
+    my $hash     = {
+        id        => $n->id,
+        directory => $n->directory->name,
+        filename  => $filename,
+    };
+    my $dir_path = $n->directory->path;
+    my $real_dir_id;
+    my $dir = $paths->{$dir_path};
+
+    if ($dir) {
+        $real_dir_id = $dir->{real_path_id} if $dir && $dir_path ne ($dir->{real_path} // '');
```

Comment:
`$dir` has been checked in the above `if ($dir)`,  maybe we do not need to check it again here.

Reply:
correct. I planned to move the above `if ($dir)` within this line but then we can't cover the else so I deleted the `if $dir &&` part within the line